src/Sylius/Bundle/CoreBundle/Security/UserImpersonator.php

```diff
@@ -51,5 +61,7 @@ public function impersonate(UserInterface $user): void
 
         $this->session->set($this->sessionTokenParameter, serialize($token));
```

Comment:
If we apply @pamil 's [comment](https://github.com/Sylius/Sylius/pull/8543/files#r137765882) we should also adjust this class

Reply:
I guess that we could think about that later, because it is a slightly different case.

Reply:
Maybe we should open an issue with this after merging? Just to not lost the track of the problem

Reply:
That's nothing related to cart storage :)